NumLib/DOF/MeshComponentMap.cpp

```diff
@@ -176,6 +176,36 @@ MeshComponentMap MeshComponentMap::getSubset(
     return MeshComponentMap(subset_dict, 1);
 }
 
+MeshComponentMap MeshComponentMap::getSubset(
+    std::vector<std::size_t> const& component_ids,
+    MeshLib::MeshSubsets const& mesh_subsets) const
+{
+    assert(component_ids.size() <= _num_components);
```

Comment:
That's what I wanted to say. Or we could even introduce something like `OGS_ASSERT(cond, msg)`.